web/src/containers/activity/CostAllocateFFP.js

```diff
@@ -95,30 +96,10 @@ CostAllocateFFP.propTypes = {
   updateActivity: PropTypes.func.isRequired
 };
 
-export const mapStateToProps = (
-  { activities: { byKey }, budget },
-  { aKey }
-) => {
-  const activity = byKey[aKey];
-  const { costAllocation } = activity;
-  const { costsByFFY } = budget.activities[aKey];
-
-  const byYearData = Object.entries(costsByFFY)
-    .filter(([year]) => year !== 'total')
-    .map(([year, costs]) => {
-      const { ffp } = costAllocation[year];
-      const ffpSelectVal = `${ffp.federal}-${ffp.state}`;
-
-      return {
-        year,
-        total: costs.total,
-        medicaidShare: costs.medicaidShare,
-        ffpSelectVal,
-        allocations: { federal: costs.federal, state: costs.state }
-      };
-    });
-
-  return { byYearData, costAllocation };
+export const makeMapStateToProps = () => {
+  const selectCostAllocateFFP = makeSelectCostAllocateFFP();
+  const mapStateToProps = (state, props) => selectCostAllocateFFP(state, props);
+  return mapStateToProps;
```

Comment:
reselect memoized functions have a cache depth of 1 - it only remembers the last arguments and return value.  If the new arguments match, it returns the old return value; otherwise, it recomputes.  This is great!  Until you need distinct state for several components simultaneously, based on props.  In that case, every call to the memoized function has unique arguments and you lose the caching behavior entirely.

To work around this, the first argument to `connect` can return a function, which allows each component to have its own `mapStateToProps` function, and thus its own version of the memoized selector. 🎉

(My rule of thumb is if your selector relies on props, you should probably use this pattern.)

Reply:
That's... whoa. I'd love to chat with you in person about this, because I feel like I sort of understand it, but not really.